solr/contrib/gcs-repository/src/test/org/apache/solr/gcs/LocalStorageGCSBackupRepository.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.solr.gcs;
+
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.contrib.nio.testing.LocalStorageHelper;
+import com.google.common.collect.Lists;
+
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.NoSuchFileException;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class LocalStorageGCSBackupRepository extends GCSBackupRepository {
+
+  protected static Storage stashedStorage = null;
+
+  @Override
+  public Storage initStorage() {
+    // LocalStorageHelper produces 'Storage' objects that track blob store state in non-static memory unique to each
+    // Storage instance.  For various components in Solr to have a coherent view of the blob-space then, they need to
+    // share a single 'Storage' object.
+    setupSingletonStorageInstanceIfNecessary();
```

Comment:
Maybe rename to 'getSingletonStorage', return the stashedStorage and set storage with the returned value?